app/controllers/organizations_controller.rb

```diff
@@ -1,21 +1,16 @@
 class OrganizationsController < ApplicationController
   after_action :verify_authorized
+  before_action :set_organization, only: %i[update generate_new_secret]
+  before_action :set_user_info
+  before_action :validate_filename, only: %i[create update]
```

Comment:
I'm not a fan of before_actions to set data https://blog.ragnarson.com/2015/09/30/problems-with-typical-rails-controllers-and-before-actions.html